### PR TITLE
Add softras examples

### PR DESCRIPTION
**Pull request by krrish94**
Add examples to demonstrate usage of the new softras core module.

1. `softras_simple_render.py`: Simple RGB image rendering
2. `softras_vertex_optimization.py`: Vertex optimization using softras
3. `softras_texture_optimization.py`: Texture optimization using softras

Examples functional. Minor TODOs remain.
* Add `argparse` to all examples.
* Clean up paths, filenames.
* Wrap `vertices` and `textures` directly into a `torch.optim.Adam` object, instead of creating an `nn.Module`.
* Write output to `.gif` file.

Signed-off-by: Krishna Murthy <krrish94@gmail.com>

### CODE REVIEW

File: examples/renderers/softras_simple_render.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
```
Comment (Caenorst) [inline examples/renderers/softras_simple_render.py:1]: Copyright should be 2020.

File: examples/renderers/softras_texture_optimization.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
```
Comment (Caenorst) [inline examples/renderers/softras_texture_optimization.py:1]: Copyright should be 2020.

File: examples/renderers/softras_vertex_optimization.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
+# Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
```
Comment (Caenorst) [inline examples/renderers/softras_vertex_optimization.py:1]: Copyright should be 2020.

**Review by Caenorst — COMMENTED** [review]
just some copyright issues

File: examples/renderers/softras_vertex_optimization.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+# Soft Rasterizer (SoftRas)
+#
+# Copyright (c) 2017 Hiroharu Kato
```
Comment (Chaitanyassr) [inline examples/renderers/softras_vertex_optimization.py:18]: these should also be changed.

**Review by Chaitanyassr — CHANGES_REQUESTED** [review]

File: examples/renderers/softras_vertex_optimization.py (hunk 4 above)
Comment (krrish94) [reply examples/renderers/softras_vertex_optimization.py:18]: These should remain as is (they are copyrights for the original impls we borrow from)

**Comment by Caenorst** [comment]
Something went wrong with the rebase, please fix.

**Comment by krrish94** [comment]
Closing, in favour of https://github.com/NVIDIAGameWorks/kaolin/pull/236

**Pull request closed without merging**